src/worker/jobs/cron/user.ts

```diff
@@ -52,4 +53,84 @@ export async function verifyUserSuspectActivity(): Promise<void> {
             }
         }
     }
+};
+
+export async function verifyDeletedAccounts(): Promise<void> {
+    const t = await sequelize.transaction();
+    try {
+        let date = new Date();
+        date.setDate(date.getDate()-15);
+
+        const users = await models.user.findAll({
+            attributes: ['address'],
+            where: {
+                deletedAt: { [Op.lt]: date },
+            },
+            include: [
+                {
+                    model: models.appUserTrust,
+                    as: 'trust'
+                }
+            ]
+        });
+
+        if(!users || !users.length) {
+            throw new Error('Users not found');
+        }
+
+        const addresses = users.map(el => el.address);
+        
+        users.forEach((user: User) => {
+            user.trust?.forEach(async (el) => {
+                await models.appUserTrust.destroy({
+                    where: {
+                        id: el.id
+                    },
+                    transaction: t
+                })
+            });
+        });
+
+        await models.beneficiary.update({
+            active: false
+        }, {
+            where: {
+                address: { [Op.in]: addresses }
+            }
+        });
+
+        await models.manager.update({
+            active: false
+        }, {
+            where: {
+                address: { [Op.in]: addresses }
+            }
+        });
+
+        await models.storyContent.destroy({
```

Comment:
What if, instead of this, we use `ON DELETE: CASCADE`, and so, we only have to delete the user? Same for `storyUserEngagement`. Even for `trust`.

Reply:
Maybe it's possible to call `models.user.destroy` with a given condition, and the database handles the remaining?

Reply:
I made some changes in migrations: storyContent and storyUserEngagement, including ON DELETE: CASCADE.
in this way, I could use only one "destroy". (on user)
but, about UserTrust table, I couldn't use the cascade. So I kept a destroy for the UserTrust table and then I do the destroy on the User table.
cascade worked fine on the app_user_through_trust table

Reply:
ahh I see. I did some search, but indeed nobody does that. Well, fine, it's done within a transaction. I think there's no issue.